packages/mui-base/src/TabsListUnstyled/TabsListUnstyled.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_useForkRef as useForkRef } from '@mui/utils';
+import { OverridableComponent } from '@mui/types';
+import composeClasses from '../composeClasses';
+import { appendOwnerState } from '../utils';
+import { getTabsListUnstyledUtilityClass } from './tabsListUnstyledClasses';
+import TabsListUnstyledProps, { TabsListUnstyledTypeMap } from './TabListUnstyledProps';
+import useTabsList from './useTabsList';
+
+const useUtilityClasses = (ownerState: { orientation: 'horizontal' | 'vertical' }) => {
+  const { orientation } = ownerState;
+
+  const slots = {
+    root: ['root', orientation],
+  };
+
+  return composeClasses(slots, getTabsListUnstyledUtilityClass, {});
+};
+
+/**
+ *
+ * Demos:
+ *
+ * - [Tabs](https://mui.com/components/tabs/)
+ *
+ * API:
+ *
+ * - [TabsListUnstyled API](https://mui.com/api/tabs-list-unstyled/)
+ */
+const TabsListUnstyled = React.forwardRef<unknown, TabsListUnstyledProps>((props, ref) => {
+  const { className, children, component, components = {}, componentsProps = {}, ...other } = props;
+
+  const { tabListRef, isRtl, orientation, getRootProps, processChildren } = useTabsList(props);
```

Comment:
So far, we usually passed in the ref to the hook instead of creating it and returning from the hook. If there's no specific reason for doing otherwise, let's keep it consistent.

Reply:
The hook needs access to the element here (via `tabListRef.current`), this is why the hook creates the ref, and let's the component pass it to the specific element.

Reply:
Isn't it similar to useButton, though? It also needs to access the element and yet it accepts the ref in props.

Reply:
Ah, I thought you meant to propagate the `ref` to the hook. I got your point, will change it.

Reply:
I am setting now the ref in the `getRootProps()` as it is done in the `useButton` hook. I am still creating the new ref in the hook, as it is not needed in the component.